platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy

```diff
@@ -50,14 +50,21 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
         """
     }
 
+    // to avoid flakiness, we need to find a JDK that is "unique" for its major version
+    private static JvmInstallationMetadata getDifferentUniqueJdk() {
+        return AvailableJavaHomes.getAvailableJvmMetadatas()
+            .findAll { it.javaMajorVersion != Jvm.current().javaVersionMajor }
+            .groupBy { it.javaMajorVersion }
+            .values().findAll { it.size() == 1 }.first().first()
+    }
+
     @Issue("https://github.com/gradle/gradle/issues/21386")
     def "emits toolchain usages for a build #configureToolchain configured toolchain for '#task' task"() {
         JvmInstallationMetadata jdkMetadata
         if (configureToolchain == "without") {
             jdkMetadata = AvailableJavaHomes.getJvmInstallationMetadata(Jvm.current())
         } else {
-            jdkMetadata = AvailableJavaHomes.getJvmInstallationMetadata(AvailableJavaHomes.differentJdk)
-
+            jdkMetadata = getDifferentUniqueJdk()
```

Comment:
❌ I don't think this change is correct. We should only be passing the specific toolchain already, so sharing a major version shouldn't matter (`withInstallations` passes the path specifically). The more relevant issue is that if the version matches the _current_ version, then there is an issue as Gradle can select the current JVM instead of a different one. But we have a much simpler existing way to do this: `AvailableJavaHomes.differentVersion`. Can you see if that works here instead?